File: docs-examples/example-java/src/test/java/io/micronaut/nats/docs/rpc/RpcUppercaseSpec.java
package io.micronaut.nats.docs.rpc;

import io.micronaut.context.annotation.Property;
import io.micronaut.test.extensions.junit5.annotation.MicronautTest;
import org.junit.jupiter.api.Test;
import reactor.core.publisher.Mono;

import static org.junit.jupiter.api.Assertions.assertEquals;

@MicronautTest
@Property(name = "spec.name", value = "RpcUppercaseSpec")
class RpcUppercaseSpec {

    @Test
    void testProductClientAndListener(ProductClient productClient) {

// tag::producer[]
assertEquals("RPC", productClient.send("rpc"));
assertEquals("HELLO", Mono.from(productClient.sendReactive("hello")).block());
// end::producer[]
    }
}

File: docs-examples/example-kotlin/src/test/kotlin/io/micronaut/nats/docs/rpc/RpcUppercaseSpec.kt
package io.micronaut.nats.docs.rpc

import io.kotest.core.spec.style.BehaviorSpec
import io.kotest.matchers.shouldBe
import io.micronaut.context.annotation.Property
import io.micronaut.test.extensions.kotest5.annotation.MicronautTest
import reactor.core.publisher.Mono

@MicronautTest
@Property(name = "spec.name", value = "RpcUppercaseSpec")
class RpcUppercaseSpec(productClient: ProductClient) : BehaviorSpec({

    given("A basic producer and consumer") {
        `when`("the message is published") {
            then("the message is consumed") {
                productClient.send("hello") shouldBe "HELLO"
                Mono.from(productClient.sendReactive("world")).block() shouldBe "WORLD"
            }
        }
    }
})
